property_tests/arbitraries/canister_methods/update_method_arb.ts

```diff
@@ -141,13 +148,22 @@ function generateSourceCode<
     functionName: string,
     paramTypes: CandidValueAndMeta<ParamType, ParamAgentType>[],
     returnType: CandidValueAndMeta<ReturnType, ReturnAgentType>,
-    callback: string
+    callback: string,
+    syntax: Syntax
 ): string {
     const paramCandidTypeObjects = paramTypes
         .map((param) => param.src.candidTypeObject)
         .join(', ');
 
     const returnCandidTypeObject = returnType.src.candidTypeObject;
 
-    return `${functionName}: update([${paramCandidTypeObjects}], ${returnCandidTypeObject}, ${callback})`;
+    if (syntax === 'functional') {
+        return `${functionName}: update([${paramCandidTypeObjects}], ${returnCandidTypeObject}, ${callback})`;
+    } else {
+        return (
+            `@update([${paramCandidTypeObjects}], ${returnCandidTypeObject})` +
```

Comment:
yes void works, yes our tests would have caught it if it was broken. The void arb is set up to return '' in this case. Since it doesn't have a corresponding IDL type '' is what we return. It works perfectly, places that need [] like funcs just put [] around '' and yields []. For where it's optional it will be @update([params], ) and then the linter clears up the extra comma and we are good to go